FSI/Displacement.C

```diff
@@ -37,13 +39,54 @@ void preciceAdapter::FSI::Displacement::initialize()
 void preciceAdapter::FSI::Displacement::write(double* buffer, bool meshConnectivity, const unsigned int dim)
 {
     /* TODO: Implement
-    * We need two nested for-loops for each patch,
-    * the outer for the locations and the inner for the dimensions.
-    * See the preCICE writeBlockVectorData() implementation.
-    */
-    FatalErrorInFunction
-        << "Writing displacements is not supported."
-        << exit(FatalError);
+     * We need two nested for-loops for each patch,
+     * the outer for the locations and the inner for the dimensions.
+     * See the preCICE writeBlockVectorData() implementation.
+     */
+
+    // Copy the displacement field from OpenFOAM to the buffer
+
+    if (this->locationType_ == LocationType::faceCenters)
+    {
+        // For every boundary patch of the interface
+        for (const label patchID : patchIDs_)
+        {
+            // Write the displacement to the preCICE buffer
+            // For every cell of the patch
+            forAll(cellDisplacement_->boundaryField()[patchID], i)
+            {
+                for (unsigned int d = 0; d < dim; ++d)
+                    buffer[i * dim + d] =
+                        cellDisplacement_->boundaryField()[patchID][i][d];
+            }
+        }
+    }
+    else if (this->locationType_ == LocationType::faceNodes)
+    {
+        WarningIn(
+            "void preciceAdapter::FSI::Displacement::write"
+            "(double* buffer, bool meshConnectivity, const unsigned int dim))")
+            << "Please be aware of issues with using 'locationType faceNodes' "
+            << "in parallel. "
+            << "See https://github.com/precice/openfoam-adapter/issues/153."
+            << endl;
```

Comment:
We actually have our own error handler, which also formats the output:

```suggestion
        DEBUG(adapterInfo(
            "Please be aware of issues with using 'locationType faceNodes' "
            "in parallel. \n"
            "See https://github.com/precice/openfoam-adapter/issues/153.",
            "warning"));
```